docs/reference/syntax-guide.rst

```diff
@@ -4,6 +4,72 @@ Syntax Guide
 
 Examples are mostly adapted from the Idris tutorial.
 
+Source File Structure
+---------------------
+
+Source files consist of:
+
+1. An optional :ref:`syntax-module-headers`.
+2. Zero or more :ref:`syntax-imports`.
+3. The file's content, which is a mix of :ref:`syntax-variables`,
+   :ref:`syntax-data-types`, :ref:`syntax-directives`, etc.
+
+.. Should that last bullet become more specific?  Terminologically I'm not even
+   certain what the top-level entities in an Idris file are properly called.
+
+For example:
+
+.. code:: idris
+
+    module MyModule   -- module header
+
+    import Data.Vect  -- an import
+
+    %default total    -- a directive
+
+    foo : Nat         -- a declaration
+    foo = 5
+
+.. _syntax-module-headers:
+
+Module Header
+~~~~~~~~~~~~~
+
+The module header appears at the top of the file.  Each file can contain only
+one module, but content within the file may be broken into multiple namespaces.
+
+Unlike most Idris declaration headers, the module header does not end in the
+keyword ``where``.
+
+A :ref:`docstring <syntax-comments>` can be placed before the module header to
+serve as module-level documentation.
+
+.. _syntax-imports:
+
+Imports
+~~~~~~~
+
+An *import* makes the names in another module available for use by the current
+module::
+
+    import Data.Vect
+
+The module's name can be used to *qualify* variables that are ambiguous, such
```

Comment:
It's not technically the module's name that's used, it's the variable's namespace. This can come from the module name, but sub-namespaces can also result from `namespace` blocks in the source code, `record` declarations, and I think a couple more places. Perhaps say that modules induce namespaces, and namespaces can be used to qualify names? Additionally, qualified names only require the _suffix_ of their namespace that makes them unique.